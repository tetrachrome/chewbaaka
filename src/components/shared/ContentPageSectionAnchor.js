/**
 * ContentPageSectionAnchor.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 07, 2020
 * Updated  : Jul 07, 2020
 */

import React from 'react';

function ContentPageSectionTitleToAnchorId(title) {
  return title.toLowerCase().replace(/\s/g , "-");
};

function ContentPageSectionAnchor(props) {
  return (
    <div id={ContentPageSectionTitleToAnchorId(props.title)} />
  );
}

export {
  ContentPageSectionTitleToAnchorId
}

export default ContentPageSectionAnchor;
