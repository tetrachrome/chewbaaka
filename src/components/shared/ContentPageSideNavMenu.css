/**
 * ContentPageSideNavMenu.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 14, 2020
 * Updated  : Jul 14, 2020
 */

div.ContentPageSideNavMenuOuterContainer {
  display: inline-block;
  font-family: Arial, Helvetica, sans-serif;
}

div.ContentPageSideNavMenuOuterContainer_DebugColor {
  background-color: beige;
}

span.ContentPageSideNavMenuHeader {
  font-size: 26px;
}

span.ContentPageSideNavMenuItem {
  color: black;
  /* background-color: blueviolet; */
  font-size: 20px;
  /* text-transform: uppercase; */
}

span.ContentPageSideNavMenuItem:hover {
  color: blue;
}
