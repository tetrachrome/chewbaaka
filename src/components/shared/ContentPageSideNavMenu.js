/**
 * ContentPageSideNavMenu.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Jul 06, 2020
 */

import React from 'react';

import { Menu } from 'semantic-ui-react'

import "semantic-ui-css/semantic.min.css";

export default class ContentPageSideNavMenu extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      // contextRef: props.contextRef,
      items : props.items,
      activeItem : props.items[0]
    }
    this.handleItemClick = this.handleItemClick.bind(this);
  }

  handleItemClick(e, { name }) {
    this.setState((prevState) => ({
      items: prevState.items,
      activeItem: name
    }));
  }

  render() {
    var menuItems = [];

    for (const [idx, item] of this.state.items.entries()) {
      menuItems.push(
        // TODO: Add routing support for menu items.
        <Menu.Item name={item} key={idx} active={this.state.activeItem === item} onClick={this.handleItemClick} />
      );
    }

    return (
      <div data-testid="ContentPageSideNavMenuComponentTestId">
        <Menu text vertical>
          <Menu.Item header>History</Menu.Item>
          {menuItems}
        </Menu>
      </div>
    );
  }
}
