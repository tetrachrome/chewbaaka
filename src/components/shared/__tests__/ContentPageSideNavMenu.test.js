/**
 * ContentPageSideNavMenu.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Jul 14, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import ContentPageSideNavMenu from '../ContentPageSideNavMenu';

import { ContentPageSectionTitleToAnchorId } from '../ContentPageSectionAnchor'

const title = "Menu Title";
const items = ["MenuItem1", "MenuItem2", "MenuItem3"];

test('renders ContentPageSideNavMenu component', () => {
  render(<ContentPageSideNavMenu title={title} items={items} />);

  const component = screen.getByTestId("ContentPageSideNavMenuComponentTestId");
  expect(component).toBeInTheDocument();

  /**
   * The text were moved inside <span></span>, and that doesn't
   * work well with getByText() apparently.
   */

  // for (const [idx, item] of items.entries()) {
  //   const menuItemElement = screen.getByText(item);
  //   expect(menuItemElement).toBeInTheDocument();
  //   expect(menuItemElement.href).toBe("http://localhost/#"+ContentPageSectionTitleToAnchorId(item));
  // }
});

test('ContentPageSideNavMenu component snapshot', () => {
  const tree = renderer
    .create(
      <ContentPageSideNavMenu title={title} items={items} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
