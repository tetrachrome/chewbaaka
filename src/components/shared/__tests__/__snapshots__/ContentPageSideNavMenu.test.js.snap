// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageSideNavMenu component snapshot 1`] = `
<div
  className="ContentPageSideNavMenuOuterContainer"
  data-testid="ContentPageSideNavMenuComponentTestId"
>
  <div
    className="ui text vertical menu"
  >
    <div
      className="header item"
      onClick={[Function]}
    >
      <span
        className="ContentPageSideNavMenuHeader"
      >
        Menu Title
      </span>
    </div>
    <a
      className="active item"
      href="#menuitem1"
      onClick={[Function]}
    >
      <span
        className="ContentPageSideNavMenuItem"
      >
        MenuItem1
      </span>
    </a>
    <a
      className="item"
      href="#menuitem2"
      onClick={[Function]}
    >
      <span
        className="ContentPageSideNavMenuItem"
      >
        MenuItem2
      </span>
    </a>
    <a
      className="item"
      href="#menuitem3"
      onClick={[Function]}
    >
      <span
        className="ContentPageSideNavMenuItem"
      >
        MenuItem3
      </span>
    </a>
  </div>
</div>
`;
